videostore/videostore.go

```diff
@@ -296,13 +296,20 @@ func (vs *videostore) Fetch(_ context.Context, r *FetchRequest) (*FetchResponse,
 		return nil, err
 	}
 	vs.logger.Debug("fetch command received")
-
 	fetchFilePath := generateOutputFilePath(
 		vs.config.Storage.OutputFileNamePrefix,
 		formatDateTimeToString(r.From),
 		"",
 		tempPath)
 
+	// Always attempt to remove the concat file after the operation.
+	// This handles error cases in Concat where it fails in the middle
+	// of writing.
+	defer func() {
+		if err := os.Remove(fetchFilePath); err != nil {
+			vs.logger.Debugf("failed to delete temporary file (%s): %v", fetchFilePath, err)
+		}
+	}()
 	if err := vs.concater.Concat(r.From, r.To, fetchFilePath); err != nil {
```

Comment:
is removing the concat.txt file a follow up?

Reply:
Seems like txt files are already being deleted properly in the `Concat` method. See here: https://github.com/viam-modules/video-store/blob/d1b64160b6992d01c26b8528ab08c0694ebee09e/videostore/concat.go#L86-L88

Reply:
Looks like txt files are already getting cleaned up properly. https://github.com/viam-modules/video-store/blob/d1b64160b6992d01c26b8528ab08c0694ebee09e/videostore/concat.go#L86-L88